Lib/fontTools/pens/ttGlyphPen.py

```diff
@@ -1,30 +1,38 @@
 from array import array
-from fontTools.misc.fixedTools import MAX_F2DOT14, otRound, floatToFixedToFloat
+from typing import Any, Optional, Tuple
+from fontTools.misc.fixedTools import MAX_F2DOT14, floatToFixedToFloat
+from fontTools.misc.loggingTools import LogMixin
+from fontTools.pens.pointPen import AbstractPointPen
 from fontTools.misc.roundTools import otRound
 from fontTools.pens.basePen import LoggingPen
-from fontTools.pens.transformPen import TransformPen
+from fontTools.pens.transformPen import TransformPen, TransformPointPen
+from fontTools.ttLib.ttFont import _TTGlyphSet
 from fontTools.ttLib.tables import ttProgram
 from fontTools.ttLib.tables._g_l_y_f import Glyph
 from fontTools.ttLib.tables._g_l_y_f import GlyphComponent
 from fontTools.ttLib.tables._g_l_y_f import GlyphCoordinates
 
 
-__all__ = ["TTGlyphPen"]
+__all__ = ["TTGlyphPen", "TTGlyphPointPen"]
 
 
-class TTGlyphPen(LoggingPen):
-    """Pen used for drawing to a TrueType glyph.
+class TTGlyphBasePen:
+    """
+    Pen used for drawing to a TrueType glyph.
 
     This pen can be used to construct or modify glyphs in a TrueType format
     font. After using the pen to draw, use the ``.glyph()`` method to retrieve
     a :py:class:`~._g_l_y_f.Glyph` object representing the glyph.
     """
 
-    def __init__(self, glyphSet, handleOverflowingTransforms=True):
-        """Construct a new pen.
+    def __init__(
+        self, glyphSet: _TTGlyphSet, handleOverflowingTransforms: bool = True
```

Comment:
This can be any kind of glyph set. I don't think it makes sense to use this (private) class as a type declaration.

Reply:
true, maybe leave it as Dict[str, Any] until we define some Drawable protocol that we can annotate this with, like https://github.com/fonttools/ufoLib2/blob/master/src/ufoLib2/typing.py